Update README to comply with Markdown Lint

santiagofn opened the pull request:
Changes were made according to https://github.com/DavidAnson/markdownlint

They include:

- Make all unordered list use asterisks
- Add empty lines before/after lists and code blocks
- Add language ref to code blocks
- Replace `<code>` with backticks
- Fix wrong link refs

santiagofn committed:
Update README to comply with Markdown Lint

Changes were made according to https://github.com/DavidAnson/markdownlint.
They include:

- Make all unordered list use asterisks
- Add empty lines before/after lists and code blocks
- Add language ref to code blocks
- Replace `<code>` with backticks
- Fix wrong link refs

reosarevok commented on README.md:
```diff
@@ -497,7 +506,8 @@ sudo docker-compose up -d
+
+1. Sample data dump is downloaded instead of full data dumps,
 2. MusicBrainz Server runs in standalone mode instead of mirror mode,
 3. development mode is enabled (but Catalyst debug),
```
```suggestion
3. Development mode is enabled (but Catalyst debug),
```

reosarevok commented on README.md:
````diff
@@ -497,7 +506,8 @@ sudo docker-compose up -d
 ```
 
 The four differences are:
````
```suggestion
The only differences are:
```
(there's seven! and we probably don't want to update the number every time we change this)

yvanzo replied:
Good catch, but it’s not exhaustive either.
```suggestion
The main differences are:
```

reosarevok commented on README.md:
```diff
@@ -271,7 +277,7 @@ Do not use it if you don't want to get your hands dirty.
-3. [Build search indexes](#build-search-indexes)
+3. [Build search indexes](#set-up-search-indexes)
    either if it has not been built
    or if it is outdated.
```
```suggestion
   if they either have not been built or are outdated.
```

And a bit above, "Make indexer goes through" -> "Make indexer go through" :)

reosarevok reviewed (commented):
This seems like an improvement, but I'll let @yvanzo confirm. I'm leaving a couple suggested improvements we could implement (in a second commit maybe?) while we're already improving this file :)

yvanzo commented on README.md:
```diff
@@ -8,33 +8,33 @@ This repo contains everything needed to run a musicbrainz mirror server with sea
+* [Update](#update)
+* [Issues](#issues)
 
 <!-- tocstop -->
```
This list is generated using the following command:
```sh
markdown-toc -i --maxdepth 3 README.md
```

It _is valid for the `sublist` style_ according to the rule [MD004](https://github.com/DavidAnson/markdownlint/blob/main/doc/md004.md).

I would suggest to rather add a configuration file for `markdownlint` to match the existing/generated style.

santiagofn replied:
You're right, I'm adding a config file. Thanks!

Although as you mentioned the alternation of symbols would be allowed using `sublist`, the previous order (1st level dash, 2nd level asterisk) would force all 1st level lists to use asterisks:

![image](https://user-images.githubusercontent.com/8375708/233538948-c1df184c-a319-4539-9f85-296c4426e031.png)

What I'm going to do is to keep the alternation but changing the order to 1st asterisk 2nd dash.

yvanzo replied:
The ToC can be generated to comply with this new style using [`markdown-toc`](https://github.com/jonschlinkert/markdown-toc)as follows:

```bash
markdown-toc -i --bullets='*' --bullets='-' --maxdepth 3 README.md
```

yvanzo commented on README.md:
````diff
@@ -188,7 +190,11 @@ Depending on your available ressources in CPU/RAM vs. bandwidth:
+
+```bash
+0 1 * * 7 YOUR_USER_NAME cd ~/musicbrainz-docker && /usr/bin/docker-compose exec -T indexer python -m sir reindex
+```
````
It’s much better to have it in a code block indeed, but it’s not a shell quote.
It should not be mislabeled as `bash` even though it makes it less colorful.

```suggestion
```crontab
0 1 * * 7 YOUR_USER_NAME cd ~/musicbrainz-docker && /usr/bin/docker-compose exec -T indexer python -m sir reindex
```
```

santiagofn replied:
Woops my bad! Will change that.

yvanzo commented on README.md:
````diff
@@ -464,10 +472,11 @@ sudo docker-compose up -d
 ```
 
 The two differences are:
-1. sample data dump is downloaded instead of full data dumps,
+
+1. Sample data dump is downloaded instead of full data dumps,
 2. MusicBrainz Server runs in standalone mode instead of mirror mode.
 
-[Build search indexes](#build-search-indexes) and
+[Build search indexes](#set-up-search-indexes) and
````
Good catch! :+1:

yvanzo reviewed (commented):
Thank you for submitting this patch!

Most of the changes are improvements, just made a few comments.

santiagofn committed:
Update markdown unorder list style to sublist

santiagofn committed:
Fix some texts & cde styles

santiagofn committed:
Enforce max line lenght (except if theyŕe code blocks)

santiagofn commented:
@reosarevok @yvanzo thanks for your feedback! I've added 3 more commits: the first 2 fix things you mentioned and the last one enforces a max line length (most of the text complied with that but there were fragments that didn't; I also added an exception for code blocks).

Let me know what you think about it and if I need to make more amends!

PS: I didn't know if you ask contributors to respect certain style on commit messages, like e.g. using a prefix, or if I must rebase all the commits into an unique commit. Let me know about that too!

yvanzo reviewed (approved):
It’s all good!

We do have guidelines for commit; See [metabrainz@guidelines/GitHub.md](https://github.com/metabrainz/guidelines/blob/master/GitHub.md). I will just squash and merge in this case, improving a little bit upon your already very detailed messages.

Thanks for your time!
